### PR TITLE
Implement is_disconnect

### PR DESCRIPTION
**Pull request by Lordshinjo**
This allows detecting that a disconnect occurred when using pool_pre_ping

### CODE REVIEW

File: sqla_vertica_python/vertica_python.py [hunk 1]
```diff
@@ -65,6 +65,18 @@ def initialize(self, connection):
         super(PGDialect, self).initialize(connection)
         self.implicit_returning = False
 
+    def is_disconnect(self, e, connection, cursor):
+        if isinstance(e, self.dbapi.Error) and connection is not None:
+            try:
+                connection = connection.connection
```
Comment (lemeryfertitta) [inline sqla_vertica_python/vertica_python.py]: I'm a bit confused by this line. `connection` is the underlying `__connection` object as far as I know. There should be no `connection.connection`.

I'm also generally unsure that this implementation is correct. From my understanding, we should just be checking the error object for text like `is closed`, e.g. https://github.com/vertica/vertica-python/blob/master/vertica_python/vertica/connection.py#L247

From [the SQLAlchemy docs](https://docs.sqlalchemy.org/en/latest/core/pooling.html#more-on-invalidation):

> all SQLAlchemy dialects include a system called is_disconnect() which will examine the contents of an exception object, including the string message and any potential error codes included with it, in order to determine if this exception indicates that the connection is no longer usable

**Review by lemeryfertitta — CHANGES_REQUESTED** [review]
Thanks for the PR! I think its good to be implementing this method, and I'd love to release it after we come up with an implementation that we both feel confident is correct.

File: sqla_vertica_python/vertica_python.py (hunk 1 above)
Comment (Lordshinjo) [reply sqla_vertica_python/vertica_python.py]: You're right about the `connection.connection`. I was looking at the implementation in `pygresql` and got confused.

About checking the error, what I'm getting in my tests is the following, which doesn't have a message at all: https://github.com/vertica/vertica-python/blob/master/vertica_python/vertica/connection.py#L458
It seems that all exceptions we care about raised in `Connection` either are caused by the connection being closed, happen before the connection is opened, or are immediately preceded by a socket close.

**Review by lemeryfertitta — APPROVED** [review]

File: sqla_vertica_python/vertica_python.py (hunk 1 above)
Comment (lemeryfertitta) [reply sqla_vertica_python/vertica_python.py]: Sounds good, thanks for the explanation. Made some slight syntactical adjustments. I'll release this tomorrow under the next minor version

**Pull request merged**